app/controllers/assignments_controller.rb

```diff
@@ -60,7 +60,12 @@ def render_test_result
   def student_interface
     @assignment = Assignment.find(params[:id])
     if @assignment.is_hidden
-      render file: "public/404.html", status: 404
+      render 'shared/http_status', formats: [:html],
+             locals: {
+                 code: '404',
```

Comment:
Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.